tasking/pulp/tasking/scheduler.py

```diff
@@ -117,38 +113,38 @@ def check_celery_processes(self):
         msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager processes '
                 'are missing for more than %d seconds') % constants.CELERY_TIMEOUT_SECONDS
         _logger.debug(msg)
-        now = ensure_tz(datetime.utcnow())
+        now = timezone.now()
         oldest_heartbeat_time = now - timedelta(seconds=constants.CELERY_TIMEOUT_SECONDS)
-        worker_list = Worker.objects.all()
-        worker_count = 0
-        resource_manager_count = 0
-        scheduler_count = 0
-
-        for worker in worker_list:
-            if worker.last_heartbeat < oldest_heartbeat_time:
-                msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
-                _logger.error(msg)
-
-                if worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
-                    worker.delete()
-                else:
-                    delete_worker(worker.name)
-            elif worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
-                scheduler_count = scheduler_count + 1
-            elif worker.name.startswith(TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME):
-                resource_manager_count = resource_manager_count + 1
+
+        for worker in Worker.objects.filter(last_heartbeat__lt=oldest_heartbeat_time):
+            msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
+            _logger.error(msg)
+
+            if worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
+                worker.delete()
             else:
-                worker_count = worker_count + 1
+                delete_worker(worker.name)
 
-        if resource_manager_count == 0:
+        worker_count = Worker.objects.exclude(
+            name__startswith=constants.RESOURCE_MANAGER_WORKER_NAME).exclude(
+            name__startswith=constants.SCHEDULER_WORKER_NAME).count()
+
+        scheduler_count = Worker.objects.filter(
+            name__startswith=constants.SCHEDULER_WORKER_NAME).count()
+
+        resource_manager_count = Worker.objects.filter(
+            name__startswith=constants.RESOURCE_MANAGER_WORKER_NAME).count()
+
+        if not resource_manager_count:
             msg = _("There are 0 pulp_resource_manager processes running. Pulp will not operate "
                     "correctly without at least one pulp_resource_mananger process running.")
             _logger.error(msg)
 
-        if scheduler_count == 0:
+        if not scheduler_count:
```

Comment:
For readability I think this was clearer with the ` == 0`. It would be good to make them all consistent.